cpp/tests/io/json_test.cpp

```diff
@@ -169,7 +169,9 @@ enum class json_test_t {
   // Run test with the existing JSON lines reader using record-orient input data
   json_lines_record_orient,
   // Run test with the nested JSON lines reader using record-orient input data
-  json_experimental_record_orient
+  json_experimental_record_orient,
+  // Run test with the nested JSON lines reader using row-orient input data
+  json_experimental_row_orient
```

Comment:
These changes clash with my changes in https://github.com/rapidsai/cudf/pull/12544/files#diff-372701f46ff1250e85be64b010783d16dbb0826f2845a412a482a1ea4ddfa023
How do you want to go about merging these two PRs?

Reply:
no problem. If either PR is merged first, I will help resolve merge issues in either PRs.
My preference would be to merge this PR first before making new reader default, so the other PR will have minimal change.

Reply:
Ideally, this enum should be replaced with 2 bools (`is_row_or_record_orient` and `is_legacy`)